druid-shell/src/platform/gtk/dialog.rs

```diff
@@ -45,6 +54,36 @@ pub(crate) fn get_file_dialog_path(
 
     dialog.set_select_multiple(options.multi_selection);
 
+    let mut found_default_filter = false;
+    if let Some(file_types) = &options.allowed_types {
+        for f in file_types {
+            let filter = file_filter(f);
+            dialog.add_filter(&filter);
+
+            if let Some(default) = &options.default_type {
+                if default == f {
+                    // Note that we're providing the same FileFilter object to
+                    // add_filter and set_filter, because gtk checks them for
+                    // identity, not structural equality.
+                    dialog.set_filter(&filter);
+                    found_default_filter = true;
+                }
+            }
+        }
+    }
+
+    if let Some(default_file_type) = &options.default_type {
+        if options.allowed_types.is_some() && !found_default_filter {
+            // It's ok to set a default file filter without providing a list of
+            // allowed filters, but it's not ok (or at least, doesn't work in gtk)
+            // to provide a default filter that isn't in the (present) list
+            // of allowed filters.
+            log::warn!("default file type not found in allowed types");
```

Comment:
This warning should be removed, because there is [an explictly documented behavior](https://github.com/xi-editor/druid/blob/af20c0906b6d62e4dc6e2568446f5bbda0de98d7/druid-shell/src/dialog.rs#L109-L110) for this case.

*If it's `None` or not present in `allowed_types` then the first entry in `allowed_types` will be used as default.*

Reply:
But it seems unreasonable for the user to provide a default that is not even available.
If this happens, it most definitely seems to be an accident and should log a warning (on all platforms), at least I think so.

Reply:
I guess that's reasonable. We can keep the warning then.

Reply:
Note that (and I can change this if it isn't desired druid behavior), if allowed_types is empty then the default type *is* allowed to be present: it applies a filter that cannot be changed.

Reply:
Just to be clear, my previous comment is about gtk's behavior.

Reply:
So if I interpret that code correctly, on Windows `Some(default_type)` is ignored if `allowed_types` is `None`. On mac the default type seems to be ignored all together.

I'm not sure what the 'correct' behavior would be (at least not the mac one).
Maybe we should allow having default be `Some` while allowed is `None`?